x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateConfig.java

```diff
@@ -0,0 +1,34 @@
+package org.elasticsearch.xpack.core.template;
+
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Pattern;
+
+public class IndexTemplateConfig {
+
+    private final String templateName;
+    private final String fileName;
+    private final String version;
+    private final String versionProperty;
+
+    public IndexTemplateConfig(String templateName, String fileName, String version, String versionProperty) {
+        this.templateName = templateName;
+        this.fileName = fileName;
+        this.version = version;
+        this.versionProperty = versionProperty;
+    }
+
+    public String getFileName() {
+        return fileName;
+    }
+
+    public String getTemplateName() {
+        return templateName;
+    }
+
+    public byte[] load() {
+        String template = TemplateUtils.loadTemplate("/" + fileName + ".json", version,
```

Comment:
Something I don't like that much about this is that we use "fileName" here, but implicitly add `.json` and prefix it with a `/`, which can be weird if someone wants to use this and provides `my_file.json` as the filename (which we'll try to load as `/my_file.json.json`)

Reply:
So we should either make the .json explicit, or change it to be `baseName` instead of `fileName`, what do you think?

Reply:
Id rather see no manipulation of the filename in this method. If you want a .json file, pass it in as such. Otherwise, if it has to be json, do some validation in the constructor instead. Thoughts @dakrone ?

Reply:
I've changed this to be the literal filename, I agree that's better.